packages/web/.storybook/main.mjs

```diff
@@ -1,13 +1,29 @@
 import { dirname, join, resolve } from 'path';
 import { mergeConfig } from 'vite';
 
-function getAbsolutePath(packageName) {
-  return dirname(resolve(__dirname, join('../..', packageName, 'package.json')));
+// This file needs to support both ESM and CJS, so we can't use `import.meta` or `__dirname` without checking
+let dir;
+try {
+  dir = __dirname;
+} catch (e) {
+  dir = import.meta.dirname;
+}
+
+function getNodeModulePath(workspace, packageName) {
+  return dirname(resolve(dir, join(workspace, 'node_modules/', packageName, 'package.json')));
+}
+
+function getRootNodeModulePath(packageName) {
+  return getNodeModulePath('../../../', packageName);
+}
+
+function getWorkspaceNodeModulePath(packageName) {
+  return getNodeModulePath('../', packageName);
```

Comment:
Having to manually resolve these `node_module` paths, and having to track where in the monorepo they get installed feels very clunky. Open to any idea on how to make this more stable and simple :pray: